The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8477171372/job/23227713738) [(plain)](https://github.com/rust-lang/rust/commit/33111d668e588b4544050655bb351131020ed09f/checks/23227713738/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:b4ffde65f46336ab88eb53be808477a3936bae11)
Download action repository 'actions/upload-artifact@v4' (SHA:5d5d22a31266ced268874388b861e4b58bb5c2f3)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.429   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.446 Collecting boolean-py==4.0
#13 0.453   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.473 Collecting chardet==5.1.0
---
#13 3.733 Building wheels for collected packages: reuse
#13 3.734   Building wheel for reuse (pyproject.toml): started
#13 4.059   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 4.060   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 4.060   Stored in directory: /tmp/pip-ephem-wheel-cache-zihmop93/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 4.063 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 4.085   Attempting uninstall: setuptools
#13 4.085     Found existing installation: setuptools 59.6.0
#13 4.087     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
---
#13 5.324   Downloading virtualenv-20.25.1-py3-none-any.whl (3.8 MB)
#13 5.501      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.8/3.8 MB 21.7 MB/s eta 0:00:00
#13 5.554 Collecting filelock<4,>=3.12.2
#13 5.561   Downloading filelock-3.13.3-py3-none-any.whl (11 kB)
#13 5.593 Collecting platformdirs<5,>=3.9.1
#13 5.600   Downloading platformdirs-4.2.0-py3-none-any.whl (17 kB)
#13 5.621 Collecting distlib<1,>=0.3.7
#13 5.630   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.642      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 48.1 MB/s eta 0:00:00
#13 5.727 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.890 Successfully installed distlib-0.3.8 filelock-3.13.3 platformdirs-4.2.0 virtualenv-20.25.1
#13 DONE 6.0s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      200640 kB
DirectMap2M:     7139328 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
    Finished `dev` profile [unoptimized] target(s) in 0.03s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/db2f9759f43167755d4eebb0a1358df9766a505e/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-db2f9759f43167755d4eebb0a1358df9766a505e-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 26.03s
##[endgroup]
fmt check
##[error]Diff in /checkout/src/tools/run-make-support/src/rustc.rs at line 1:
 use std::env;
-use std::process::{Command, Output};
 use std::path::PathBuf;
+use std::process::{Command, Output};
 
 
 use crate::{handle_failed_output, tmp_dir};
 
Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/src/tools/tidy/src/features/version.rs" "/checkout/src/bootstrap/src/bin/rustdoc.rs" "/checkout/src/bootstrap/src/bin/sccache-plus-cl.rs" "/checkout/src/bootstrap/src/bin/main.rs" "/checkout/src/bootstrap/src/bin/rustc.rs" "/checkout/src/bootstrap/build.rs" "/checkout/src/tools/jsondoclint/src/json_find.rs" "/checkout/src/tools/jsondoclint/src/json_find/tests.rs" "/checkout/src/tools/jsondoclint/src/main.rs" "/checkout/src/tools/jsondoclint/src/item_kind.rs" "/checkout/src/tools/jsondoclint/src/validator/tests.rs" "/checkout/src/tools/jsondoclint/src/validator.rs" "/checkout/src/tools/html-checker/main.rs" "/checkout/src/tools/collect-license-metadata/src/path_tree.rs" "/checkout/src/tools/collect-license-metadata/src/reuse.rs" "/checkout/src/tools/collect-license-metadata/src/main.rs" "/checkout/src/tools/collect-license-metadata/src/licenses.rs" "/checkout/src/tools/llvm-bitcode-linker/src/target.rs" "/checkout/src/tools/llvm-bitcode-linker/src/lib.rs" "/checkout/src/tools/llvm-bitcode-linker/src/opt.rs" "/checkout/src/tools/llvm-bitcode-linker/src/linker.rs" "/checkout/src/tools/llvm-bitcode-linker/src/bin/llvm-bitcode-linker.rs" "/checkout/src/tools/bump-stage0/src/main.rs" "/checkout/src/tools/suggest-tests/src/tests.rs" "/checkout/src/tools/suggest-tests/src/lib.rs" "/checkout/src/tools/suggest-tests/src/static_suggestions.rs" "/checkout/src/tools/suggest-tests/src/main.rs" "/checkout/src/tools/suggest-tests/src/dynamic_suggestions.rs" "/checkout/compiler/rustc_driver_impl/src/session_diagnostics.rs" "/checkout/compiler/rustc_driver_impl/src/lib.rs" "/checkout/compiler/rustc_driver_impl/src/pretty.rs" "/checkout/compiler/rustc_driver_impl/src/print.rs" "/checkout/compiler/rustc_driver_impl/src/signal_handler.rs" "/checkout/compiler/rustc_driver_impl/src/args.rs" "/checkout/src/tools/unicode-table-generator/src/case_mapping.rs" "/checkout/src/tools/unicode-table-generator/src/unicode_download.rs" "/checkout/src/tools/unicode-table-generator/src/range_search.rs" "/checkout/src/tools/unicode-table-generator/src/main.rs" "/checkout/src/tools/unicode-table-generator/src/cascading_map.rs" "/checkout/src/tools/unicode-table-generator/src/raw_emitter.rs" "/checkout/src/tools/unicode-table-generator/src/skiplist.rs" "/checkout/src/tools/tier-check/src/main.rs" "/checkout/src/tools/error_index_generator/main.rs" "/checkout/compiler/rustc_interface/src/tests.rs" "/checkout/compiler/rustc_interface/src/queries.rs" "/checkout/compiler/rustc_interface/src/lib.rs" "/checkout/compiler/rustc_interface/src/callbacks.rs" "/checkout/compiler/rustc_interface/src/errors.rs" "/checkout/compiler/rustc_interface/src/interface.rs" "/checkout/compiler/rustc_interface/src/proc_macro_decls.rs" "/checkout/compiler/rustc_interface/src/passes.rs" "/checkout/compiler/rustc_interface/src/util.rs" "/checkout/src/tools/run-make-support/src/rustdoc.rs" "/checkout/src/tools/run-make-support/src/lib.rs" "/checkout/src/tools/run-make-support/src/run.rs" "/checkout/src/tools/run-make-support/src/rustc.rs" "/checkout/src/tools/linkchecker/main.rs" "/checkout/compiler/rustc_feature/src/tests.rs" "/checkout/compiler/rustc_feature/src/removed.rs" "/checkout/compiler/rustc_feature/src/accepted.rs" "/checkout/compiler/rustc_feature/src/lib.rs" "/checkout/compiler/rustc_feature/src/unstable.rs" "/checkout/compiler/rustc_feature/src/builtin_attrs.rs" "/checkout/src/tools/tidy/src/features/version/tests.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Fri Mar 29 04:30:06 UTC 2024
  network time: Fri, 29 Mar 2024 04:30:06 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/tools/run-make-support/src/rustc.rs
+++ b/src/tools/run-make-support/src/rustc.rs
@@ -104,6 +104,20 @@ impl Rustc {
         self
     }
 
+    /// Specify the crate type.
+    pub fn crate_type(&mut self, crate_type: &str) -> &mut Self {
+        self.cmd.arg("--crate-type");
+        self.cmd.arg(crate_type);
+        self
+    }
+
+    /// Specify the edition year.
+    pub fn edition(&mut self, edition: &str) -> &mut Self {
+        self.cmd.arg("--edition");
+        self.cmd.arg(edition);
+        self
+    }
+
     /// Generic command arguments provider. Use `.arg("-Zname")` over `.arg("-Z").arg("arg")`.
     /// This method will panic if a plain `-Z` or `-C` is passed, or if `-Z <name>` or `-C <name>`
     /// is passed (note the space).

--- a/src/tools/tidy/src/allowed_run_make_makefiles.txt
+++ b/src/tools/tidy/src/allowed_run_make_makefiles.txt
@@ -31,7 +31,6 @@ run-make/compiler-rt-works-on-mingw/Makefile
 run-make/compressed-debuginfo/Makefile
 run-make/const-prop-lint/Makefile
 run-make/const_fn_mir/Makefile
-run-make/core-no-fp-fmt-parse/Makefile
 run-make/core-no-oom-handling/Makefile
 run-make/crate-data-smoke/Makefile
 run-make/crate-hash-rustc-version/Makefile

--- a/tests/run-make/core-no-fp-fmt-parse/Makefile
+++ b/tests/run-make/core-no-fp-fmt-parse/Makefile
@@ -1,4 +1,0 @@
-include ../tools.mk
-
-all:
-	$(RUSTC) --edition=2021 -Dwarnings --crate-type=rlib ../../../library/core/src/lib.rs --cfg no_fp_fmt_parse

--- a/tests/run-make/core-no-fp-fmt-parse/rmake.rs
+++ b/tests/run-make/core-no-fp-fmt-parse/rmake.rs
@@ -1,0 +1,17 @@
+// This test checks that the core library of Rust can be compiled without enabling
+// support for formatting and parsing floating-point numbers.
+
+extern crate run_make_support;
+
+use run_make_support::rustc;
+use std::path::PathBuf;
+
+fn main() {
+    rustc()
+        .edition("2021")
+        .arg("-Dwarnings")
+        .crate_type("rlib")
+        .input("../../../library/core/src/lib.rs")
+        .cfg("no_fp_fmt_parse")
+        .run();
+}

